src/chainstate/stacks/index/storage.rs

```diff
@@ -1310,76 +971,77 @@ impl TrieFileStorage {
         // after.  Turns out rename(2) isn't crash-consistent, and turns out syscalls can get
         // reordered.
         if let Some((ref bhh, ref mut trie_ram)) = self.last_extended.take() {
-            let block_path_tmp = TrieFileStorage::block_path_tmp(&self.dir_path, bhh);
-            let (block_path, real_bhh) = match final_bhh {
-                Some(real_bhh) => {
-                    if *real_bhh != *bhh {
-                        self.block_retarget(bhh, real_bhh)?;
-                        assert_eq!(self.block_map.find_id(real_bhh), Some(trie_ram.identifier));
+            trace!("Buffering block flush started.");
+            let mut buffer = Cursor::new(Vec::new());
+            trie_ram.dump(&mut buffer, bhh)?;
```

Comment:
We're _not_ streaming directly to the blob, but buffering to RAM first?

Reply:
Only on writes -- reads are streamed. I don't think this adds that much overhead, but can update this to write a zero blob and then stream the write.

Reply:
Maybe not in this PR (bird in the hand, two in the bush, etc), but let's do this if we get complaints  about memory usage.